src/battle_util.c

```diff
@@ -3484,7 +3484,10 @@ u8 AtkCanceller_UnableToUseMove(u32 moveType)
             gBattleStruct->atkCancellerTracker++;
             break;
         case CANCELLER_PARALYSED: // paralysis
-            if (!gBattleStruct->isAtkCancelerForCalledMove && (gBattleMons[gBattlerAttacker].status1 & STATUS1_PARALYSIS) && !RandomPercentage(RNG_PARALYSIS, 75))
+            if (!gBattleStruct->isAtkCancelerForCalledMove
+             && gBattleMons[gBattlerAttacker].status1 & STATUS1_PARALYSIS
+             && !RandomPercentage(RNG_PARALYSIS, 75)
+             && (GetBattlerAbility(gBattlerAttacker) != ABILITY_MAGIC_GUARD && B_MAGIC_GUARD >= GEN_4))
```

Comment:
Magic Guard check should go before the random roll, so that the roll never happens if the Magic Guard would prevent the move prevention.

Reply:
I thought about this but isn't it more likely the random roll fails more often then a user having magic guard? You would basically do that Magic Guard check 75% of the time for nothing.

Reply:
Wouldn't a call to `RandomPercentage` be more expensive than checking the ability 3 times?
Or is it irrelevant?

Reply:
It is more expensive. My thought process is that the magic guard check will be true 99% of the time which means the random call will be executed regardless pretty much every time. But since the random call would fail 75% of the time we avoid running the magic guard check in those cases and run it only 25% of the time.

Reply:
Obviously this is only true if we assume that magic guard is a very rare ability which it is in vanilla.

Reply:
Maybe I'm wrong if the random call is really expensive and the ability check really cheap though

Reply:
let's just change the order since the ability check is cheap and I don't know how expensive the random call is

Reply:
changed